### ISSUE TITLE
active_dims not working as expected in >=2.0.0

### ISSUE THREAD

**Issue by janskaar** [GPflow/GPflow#1445]
# Bug / performance issue / build issue

After upgrading to v2.0.1 I found that specifying active_dims seemingly doesn't work:

## To reproduce

**Minimal, reproducible example**

```import gpflow as gpf
import numpy as np

a = np.arange(6).reshape((-1,2)).astype(np.float64)
a[:,1] = a[:,1] / 5

k1 = gpf.kernels.RBF(active_dims=[0]) + gpf.kernels.RBF(active_dims=[1])
k2 = gpf.kernels.RBF() + gpf.kernels.RBF()

cov1 = k1.K(a, a).numpy()
cov2 = k2.K(a, a).numpy()

print(cov1)
print(cov2)
```

**Stack trace, or error message**
Using GPflow v2.0.0rc1 this ouputs 
```
[[2.         1.05845163 0.7264845 ]
 [1.05845163 2.         1.05845163]
 [0.7264845  1.05845163 2.        ]]
[[2.00000000e+00 2.49860424e-01 4.87191728e-04]
 [2.49860424e-01 2.00000000e+00 2.49860424e-01]
 [4.87191728e-04 2.49860424e-01 2.00000000e+00]]
```

Using GPflow v2.0.0 and 2.0.1 this outputs 
```
[[2.00000000e+00 2.49860424e-01 4.87191728e-04]
 [2.49860424e-01 2.00000000e+00 2.49860424e-01]
 [4.87191728e-04 2.49860424e-01 2.00000000e+00]]
[[2.00000000e+00 2.49860424e-01 4.87191728e-04]
 [2.49860424e-01 2.00000000e+00 2.49860424e-01]
 [4.87191728e-04 2.49860424e-01 2.00000000e+00]]
```
i.e. specifying active_dims doesn't have any effect.

**Comment by st--** [comment]
Ah, I believe this isn't technically a bug but how you use the kernel. In GPflow 2 (including 2.0.0rc1!), to compute the kernel matrix, you should simply *call* the kernel object, i.e.
```diff
-cov1 = k1.K(a, a).numpy()
-cov2 = k2.K(a, a).numpy()
+cov1 = k1(a, a).numpy()
+cov2 = k2(a, a).numpy()
```
If you make this change, GPflow 2.0.0 and 2.0.1 give you the same answer you were expecting.
Have you come across any gpflow 2 example that explicitly call `kernel.K()` instead of `kernel()`? What could we do to make this clearer in our documentation? :thinking:

**Comment by st--** [comment]
Maybe we should deprecate K and K_diag and turn them into private methods instead, as they are the methods you should *implement* when writing your own kernels, not ones you should call directly when *using* a kernel

**Comment by janskaar** [comment]
Aah, I see. I don't remember having come across any recent examples using `kernel.K()`, I must not have picked up on this change since it still worked in 2.0.0rc1 (I think it `kernel.K()` was used in 1.x?) Thanks for clearing up my misunderstanding.

**Comment by st--** [comment]
@janskaar, yes, in GPflow 1.x it was `kernel.K()`, and for *most* kernels, `kernel.K()` and `kernel()` behaved equivalently in 2.0.0rc1. However, behaviour wasn't consistent across all kernels, which is why we moved the preslicing into the kernel's `__call__()` method in PR #1235.
